ledger/ledger-api-bench-tool/src/main/scala/com/daml/ledger/api/benchtool/metrics/MetricRegistryOwner.scala

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.api.benchtool.metrics
+
+import scala.concurrent.duration.Duration
+import java.util.concurrent.TimeUnit
+import com.codahale.metrics.Slf4jReporter.LoggingLevel
+import com.codahale.metrics.{MetricRegistry, Reporter, Slf4jReporter}
+import com.daml.ledger.resources.{Resource, ResourceContext, ResourceOwner}
+import com.daml.metrics.MetricsReporter
+import com.daml.resources
+
+import scala.concurrent.Future
+
+class MetricRegistryOwner(
+    reporter: MetricsReporter,
+    reportingInterval: Duration,
+) extends ResourceOwner[MetricRegistry] {
+  override def acquire()(implicit
+      context: ResourceContext
+  ): resources.Resource[ResourceContext, MetricRegistry] = {
+    val registry = new MetricRegistry
+    for {
+      slf4JReporter <- acquire(newSlf4jReporter(registry))
+      _ <- acquire(reporter.register(registry))
+        .map(_.start(reportingInterval.toSeconds, TimeUnit.SECONDS))
+      // Trigger a report to the SLF4J logger on shutdown.
+      _ <- Resource(Future.successful(slf4JReporter))(reporter =>
+        Future.successful(reporter.report())
+      )
+    } yield registry
+  }
+
+  private def newSlf4jReporter(registry: MetricRegistry): Slf4jReporter =
+    Slf4jReporter
+      .forRegistry(registry)
+      .convertRatesTo(TimeUnit.SECONDS)
+      .convertDurationsTo(TimeUnit.MILLISECONDS)
+      .withLoggingLevel(LoggingLevel.DEBUG)
```

Comment:
Can we have these source from a `logback.xml`?

Reply:
I've redirected logging to the global logger so that changing the root log level will allow to filter out the report.
I'll try to add more configuration in subsequent PRs.